Mathlib/Topology/Separation.lean

```diff
@@ -129,6 +147,54 @@ theorem separatedNhds_iff_disjoint {s t : Set X} : SeparatedNhds s t ↔ Disjoin
 
 alias ⟨SeparatedNhds.disjoint_nhdsSet, _⟩ := separatedNhds_iff_disjoint
 
+/-- `SeparatingCover`s can be useful witnesses for `SeparatedNhds`. -/
+def SeparatingCover : Set X → Set X → Prop := fun s t ↦
+  ∃ u : ℕ → Set X, s ⊆ ⋃ n, u n ∧ ∀ n, IsOpen (u n) ∧ Disjoint (closure (u n)) t
+
+/-- Used to prove that a regular topological space with Lindelöf topology is a normal space,
+and (todo) a perfectly normal space is a completely normal space. -/
+theorem separating_covers_iff_separated_nhds {h k : Set X} :
+    SeparatingCover h k ∧ SeparatingCover k h ↔ SeparatedNhds h k := by
+  constructor
+  · rintro ⟨⟨u, u_cov, u_props⟩, ⟨v, v_cov, v_props⟩⟩
+    have open_lemma : ∀ (u₀ a : ℕ → Set X), (∀ n, IsOpen (u₀ n)) →
+      IsOpen (⋃ n, u₀ n \ closure (a n)) := fun _ _ u₀i_open ↦
+      isOpen_iUnion (fun i ↦ IsOpen.sdiff (u₀i_open i) (isClosed_closure))
+    have cover_lemma : ∀ (h₀ : Set X) (u₀ v₀ : ℕ → Set X),
+        (h₀ ⊆ ⋃ n, u₀ n) → (∀ n, Disjoint (closure (v₀ n)) h₀) →
+        (h₀ ⊆ ⋃ n, u₀ n \ closure (⋃ m ∈ {m | m ≤ n}, v₀ m)) :=
```

Comment:
This is really not a nice looking spelling. If we are missing lemmas to write `⋃ m ≤ n, v₀ m` then we should add them instead of using the ugly spelling.

Reply:
Indeed I tried `⋃ m ≤ n, v₀ m` but struggled. Will take another stab at this (and add a lemma if needed).

Reply:
Accomplished with https://github.com/leanprover-community/mathlib4/compare/0f84d6b0c33c428f4a59d90497e3a430ff69acdf..7d76efc808080048d5b3e844f76c465ab11d6989 - I suspect there's a better name than `closure_iUnion₂_of_finite`. And the CI failed so I'm addressing that now.